proteuslib/tests/test_remineralization.py

```diff
@@ -0,0 +1,1543 @@
+        check.fs.unit.inlet.mole_frac_comp[0, "HCO3_-"].fix( nahco3 + 2*cahco3 )
+        check.fs.unit.inlet.mole_frac_comp[0, "CO3_2-"].fix( caco3 )
+        check.fs.unit.inlet.mole_frac_comp[0, "CO2"].fix( 0.0005 )
+
+        sum = 0
+        for i in check.fs.unit.inlet.mole_frac_comp:
+            if i[1] != "H2O":
+                sum += value(check.fs.unit.inlet.mole_frac_comp[i[0], i[1]])
+
+        check.fs.unit.inlet.mole_frac_comp[0, "H2O"].fix( 1-sum )
+
+        check.fs.unit.inlet.pressure.fix(101325.0)
+        check.fs.unit.inlet.temperature.fix(298.)
+        check.fs.unit.outlet.temperature.fix(298.)
+        check.fs.unit.inlet.flow_mol.fix(10)
+
+        #Custom eps factors
+        eps = 1e-30
+        check.fs.thermo_params.reaction_H2O_Kw.eps.value = eps
+        check.fs.thermo_params.reaction_H2CO3_Ka1.eps.value = eps
+        check.fs.thermo_params.reaction_H2CO3_Ka2.eps.value = eps
+        check.fs.thermo_params.reaction_CO2_to_H2CO3.eps.value = eps
+
+        solver.options['bound_push'] = 1e-20
+        solver.options['mu_init'] = 1e-6
+        solver.options['max_iter'] = 1000
+        check.fs.unit.initialize(optarg=solver.options)
+
+        results = solver.solve(check, tee=True)
+        assert results.solver.termination_condition == TerminationCondition.optimal
+        assert results.solver.status == SolverStatus.ok
+
+        #NOTE: The tolerances below are intentionally loosened, due to the inexact nature
+        #       of the conversion between true and apparent species
+        total_molar_density = value(check.fs.unit.control_volume.properties_out[0.0].dens_mol_phase['Liq'])/1000
+        pH = -value(log10(check.fs.unit.outlet.mole_frac_comp[0, "H_+"]*total_molar_density))
+        pOH = -value(log10(check.fs.unit.outlet.mole_frac_comp[0, "OH_-"]*total_molar_density))
+
+        assert pytest.approx(8.182189388725178, rel=1e-5) == pH
+        assert pytest.approx(5.818979926705361, rel=1e-5) == pOH
+
+        # Calculate total hardness
+        TH = 2*value(check.fs.unit.control_volume.properties_out[0.0].conc_mol_phase_comp[('Liq', 'Ca_2+')])/1000
+        TH = TH*50000
+        assert pytest.approx(59.58239143431564, rel=1e-5) == TH
+
+        # Calculating carbonate alkalinity to determine the split of total hardness
+        CarbAlk = 2*value(check.fs.unit.control_volume.properties_out[0.0].conc_mol_phase_comp[('Liq', 'CO3_2-')])/1000
+        CarbAlk += value(check.fs.unit.control_volume.properties_out[0.0].conc_mol_phase_comp[('Liq', 'HCO3_-')])/1000
+        CarbAlk = 50000*CarbAlk
+        assert pytest.approx(160.3367559866467, rel=1e-5) == CarbAlk
+
+        # Non-Carbonate Hardness only exists if there is excess hardness above alkalinity
+        if TH <= CarbAlk:
+            NCH = 0
+        else:
+            NCH = TH - CarbAlk
+        CH = TH - NCH
+        assert pytest.approx(TH, rel=1e-5) == CH
+
+    @pytest.fixture(scope="class")

```

Comment:
Just to help organise things a little better, I would suggest making this a separate test class. I would also suggest moving the declaration of the config dicts for these test to just immediately before here.

The reason being it helps group information together better to help the reader understand - when I started reading I was wondering why there were two different sets of configs and how you intended to use each of them.